### PR TITLE
Add description functions

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
This PR adds:
- Initial interpreter, to process description block
- Error handling for interpreter
- NASL built-in functions for description block

SC-699

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

File: rust/README.md [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 This multi cargo workspace contains
 - nasl-syntax - to parse nasl for further processing
+- nasl-built-in - to call built-in functions
```
Comment (Kraemii) [inline rust/README.md:5]: change

File: rust/nasl-interpreter/src/interpreter.rs [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    fn add_global(&mut self, name: &str, value: ContextType);
+    /// Adds a positional function parameter to the context
+    fn add_postitional(&mut self, value: ContextType);
+    /// Returns the value of a named parameter/variable or None if it does not exist
+    fn get_named(&self, name: &str) -> Option<&ContextType>;
+    /// Returns the value of a positional parameter or None if it does not exist
+    fn get_positional(&self, pos: usize) -> Option<&ContextType>;
+    /// Return a new Context, which contains the global variables of the current Context
+    fn globals_copy(&self) -> Box<dyn NaslContext>;
+}
```
Comment (Kraemii) [inline rust/nasl-interpreter/src/interpreter.rs:37]: use block depth for identifiers

File: rust/nasl-interpreter/src/interpreter.rs [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+            NaslValue::Boolean(boolean) => boolean,
+            NaslValue::Null => false,
+            NaslValue::Number(number) => number != 0,
+        }
+    }
+
+    /// Interpetes a Statement
+    pub fn resolve(&mut self, statement: Statement) -> Result<NaslValue, InterpetError> {
+        match statement {
+            Array(_,_ ) => Ok(NaslValue::Null),
```
Comment (Kraemii) [inline rust/nasl-interpreter/src/interpreter.rs:123]: change to todo macro

File: rust/nasl-interpreter/src/interpreter.rs [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+                }
+            },
+            Variable(_) => Ok(NaslValue::Null),
+            Call(function_name, parameters) => {
+                // Get the function, if it exists
+                let name = &self.code[function_name.range()];
+                match lookup(name){
+                    // Built-In Function
+                    Some(function) => {
+                        let mut function_context = self.context.globals_copy();
```
Comment (Kraemii) [inline rust/nasl-interpreter/src/interpreter.rs:147]: don't need a copy

**Review by nichtsfrei — APPROVED** [review]

**Review by ArnoStiefvater — APPROVED** [review]

**Pull request merged**